cli/flox/src/commands/general.rs

```diff
@@ -209,3 +209,87 @@ pub(super) fn update_config<V: Serialize>(
             }
     Ok(())
 }
+
+/// Parse a TOML key from a string, quoting any segments where necessary, so
+/// that a user doesn't need to understand the intricacies of TOML.
+fn parse_toml_key(key: &str) -> Result<Vec<Key>, TomlError> {
+    let normalized_key = key
+        .split('.')
+        .map(|segment| {
+            let quoting_not_needed = segment
+                .chars()
+                .all(|c| c.is_alphanumeric() || c == '_' || c == '-');
+            let contains_some_quotes = segment.contains('"') || segment.contains('\'');
+
+            if quoting_not_needed || contains_some_quotes {
+                segment.to_string()
+            } else {
+                format!("'{}'", segment)
+            }
+        })
+        .collect::<Vec<_>>()
+        .join(".");
+
+    Key::parse(&normalized_key)
+}
+
+#[cfg(test)]
+mod tests {
+    use pretty_assertions::assert_eq;
+
+    use super::*;
+
+    #[test]
+    fn parse_toml_key_no_quoting_needed() {
+        let key = "trusted_environments.foo.bar";
+        let parsed = parse_toml_key(key).unwrap();
+        assert_eq!(parsed, vec!["trusted_environments", "foo", "bar"]);
+    }
+
+    #[test]
+    fn parse_toml_key_adds_quoting() {
+        let key = "trusted_environments.foo/bar";
+        let parsed = parse_toml_key(key).unwrap();
+        assert_eq!(parsed, vec!["trusted_environments", "foo/bar"]);
+    }
+
+    #[test]
+    fn parse_toml_key_already_single_quoted() {
+        let key = "trusted_environments.'foo/bar'";
+        let parsed = parse_toml_key(key).unwrap();
+        assert_eq!(parsed, vec!["trusted_environments", "foo/bar"]);
+    }
+
+    #[test]
+    fn parse_toml_key_already_double_quoted() {
+        let key = r#"trusted_environments."foo/bar""#;
+        let parsed = parse_toml_key(key).unwrap();
+        assert_eq!(parsed, vec!["trusted_environments", "foo/bar"]);
+    }
+
+    #[test]
+    fn parse_toml_key_stray_single_quote() {
+        let key = "trusted_environments.foo'bar";
+        let err = parse_toml_key(key).unwrap_err();
+        assert_eq!(err.to_string(), indoc! {r#"
+            TOML parse error at line 1, column 25
+              |
+            1 | trusted_environments.foo'bar
+              |                         ^
+
+        "#});
+    }
+
+    #[test]
+    fn parse_toml_key_stray_double_quote() {
+        let key = r#"trusted_environments.foo"bar"#;
+        let err = parse_toml_key(key).unwrap_err();
+        assert_eq!(err.to_string(), indoc! {r#"
+            TOML parse error at line 1, column 25
+              |
+            1 | trusted_environments.foo"bar
+              |                         ^
+
+        "#});
+    }
```

Comment:
note nonblocking: seems like proptest could be good for testing this

Reply:
For just the error cases, or all of them? Can you describe why and what you had in mind, e.g. generating the whole key path or individual parts of it?

Reply:
Once again I find myself liking the idea but in practice it feels too much like tautology and unnecessary work so I'm going to defer for now.